### ISSUE TITLE
Considering multiple metallicities

### ISSUE THREAD

**Issue by themiyan** [cschreib/fastpp#7]
can I follow up on the new build regarding Z? :)

METAL = [0.004, 0.02, 0.05, 0.008]

In the past commit this would use all SSPs in this array, but now it seems it uses only the first value specified in the array.

**Comment by themiyan** [comment]
@cschreib response
FYI: It's better to open a new issue for un-related problems ;)

As for your problem, I just tried running a fit with an array of metallicities and they were all considered in the fit. Are you sure the METAL option is not being re-defined twice in the param file? When running the program, do you see a line like this printed in the terminal:

note: fitting a grid of XXX templates (ntau=XX,nmetal=4,nage=XX,nav=XX,nz=XXX)
... with nmetal=4 ?

**Comment by themiyan** [comment]
yeap it does this: 

with 
`METAL          = [0.004, 0.02, 0.05, 0.008]  `
 I get:
`note: fitting a grid of 1299056 templates (ntau=4,nmetal=4,nage=11,nav=61,nz=121)`

but best fit outputs are all for Z= 0.004

If I change to:
`METAL          = [0.02, 0.004 , 0.05, 0.008] `
I get 0.02 for most of the outputs with a lot of nans

and if I put 0.008 in front almost all values are 0.008 with a small fraction being different. So it seems the order how `METAL` is defined matters? 

The ZFIRE config I'm running is here: https://www.dropbox.com/sh/obka9ir6ezv2sb4/AACQRg5BtiKdaSQc5CSZlPIza?dl=0

**Comment by cschreib** [comment]
Ok, that does sound weird. I will look into this as soon as I have time!

**Comment by cschreib** [comment]
Ah, I think I got it! Didn't take as long as I feared :) The bug happens when the ```METAL``` list is not sorted and ```BEST_FROM_SIM=1```. In the tests I did initially, I had ```BEST_FROM_SIM=0``` (the default), so I didn't experience the bug. I've uploaded the fix to the master branch, let me know if that doesn't solve the issue.

I've also added another change to the way the output values are estimated from MC simulations, which should only impact small grids (mostly for ```metal```, but also ```tau``` to a lesser extent). This should give more correct output values, i.e., not biased toward the grid edges.
